packages/protocol/test/ERC1724/ZkAssetMintable.js

```diff
@@ -1,264 +1,228 @@
-
-            // total inputs != total outputs - ace.mint throws
-            const noteValues = [50, 0, 30, 30];
-            const notes = await Promise.all(
-                aztecAccounts.map(({ publicKey }, i) => {
-                    return note.create(publicKey, noteValues[i]);
-                }),
+            const zkAssetMintable = await ZkAssetMintable.new(
+                ace.address,
+                erc20.address,
+                scalingFactor,
+                canAdjustSupply,
+                canConvert,
+                {
+                    from: accounts[0],
+                },
             );
-
-            zkAssetMintable = await ZkAssetMintable.new(ace.address, erc20.address, scalingFactor, canAdjustSupply, canConvert, {
-                from: accounts[0],
-            });
-
-            const newTotalMinted = notes[0];
-            const oldTotalMinted = await note.createZeroValueNote();
-            const adjustedNotes = notes.slice(2, 4);
-
-            // Minting two AZTEC notes, worth 30 and 30
-            proofs[0] = proof.mint.encodeMintTransaction({
-                newTotalMinted,
-                oldTotalMinted,
-                adjustedNotes,
-                senderAddress: zkAssetMintable.address,
-            });
-            await truffleAssert.reverts(zkAssetMintable.confidentialMint(MINT_PROOF, proofs[0].proofData));
+            const sender = zkAssetMintable.address;
+            const newMintCounterValue = 50;
+            const mintedNoteValues = [30, 30];
+            const { zeroMintCounterNote, newMintCounterNote, mintedNotes } = await getCustomMintNotes(
+                newMintCounterValue,
+                mintedNoteValues,
+            );
+            const proof = new MintProof(zeroMintCounterNote, newMintCounterNote, mintedNotes, sender);
+            const data = proof.encodeABI();
+            await truffleAssert.reverts(zkAssetMintable.confidentialMint(MINT_PROOF, data));
         });
 
-        it('should fail if flag set to FALSE', async () => {
-            const proofs = [];
+        it('should fail to mint if canAdjustSupply flag set to FALSE', async () => {
             const canAdjustSupply = false;
             const canConvert = true;
-
-            const aztecAccounts = [...new Array(4)].map(() => secp256k1.generateAccount());
-            const noteValues = [50, 0, 30, 20]; // note we do not use this third note, we create fixed one
-            const notes = await Promise.all(
-                aztecAccounts.map(({ publicKey }, i) => {
-                    return note.create(publicKey, noteValues[i]);
-                }),
+            const zkAssetMintable = await ZkAssetMintable.new(
+                ace.address,
+                erc20.address,
+                scalingFactor,
+                canAdjustSupply,
+                canConvert,
+                {
+                    from: accounts[0],
```

Comment:
Not necessary to specify `from` here.

Reply:
Agreed, but in some cases the address of the sender is significant. A proof may fail if an incorrect sender address was used, so I've taken to making the `from` field explicit